File: backend/app/src/models/locations/Planet.py
from ..blueprints.Base import Base



class Planet(Base):
    def __init__(self, name: str, base_type = "planet", tile_id: int = 0, settlements: list = []):
        super().__init__(name=name, base_type=base_type, tile_id=tile_id, settlements=settlements)

File: backend_refactoring/CSharpApi/Models/AstronomicalObjects/Planet.cs
using CSharpApi.Data;
using CSharpApi.Models.BluePrints;
using CSharpApi.Models.BluePrints.Locations;
using Microsoft.EntityFrameworkCore;
using System.Runtime.InteropServices;
using System.Xml.Linq;

namespace CSharpApi.Models.AstronomicalObjects
{
    public class Planet : ICelestialObject
    {
        public string Name { get; set; } = string.Empty;

        string ICelestialObject.Name => Name;

        private string BaseType { get; set; } = "planet";

        private int TileId { get; set; } = 0;

        private List<Settlement> Settlements { get; set; } = [];

        public Planet(string name, string baseType, int tileId, List<Settlement> settlements) {
            Name = name;
            BaseType = baseType;
            TileId = tileId;
            Settlements = settlements;
        }
    }
}